aptos-move/aptos-vm/src/lib.rs

```diff
@@ -153,14 +153,15 @@ pub trait VMValidator {
 }
 
 /// This trait describes the VM's execution interface.
-pub trait VMExecutor: Send + Sync {
-    // NOTE: At the moment there are no persistent caches that live past the end of a block (that's
-    // why execute_block doesn't take &self.)
-    // There are some cache invalidation issues around transactions publishing code that need to be
-    // sorted out before that's possible.
+pub trait VMBlockExecutor: Send + Sync {
+    /// Be careful if any state is kept in VMBlockExecutor, as all validations are implementers responsibility
```

Comment:
Let's move this comment to the trait definition? Also `VM's execution interface.` is no longer correct above.

typo: `implementers` -> `implementer's`

Reply:
this comment applies to implementation of new , so having it on new function seems reasonable? but we can change it later if needed.